src/components/publishing/email_signup.tsx

```diff
@@ -0,0 +1,120 @@
+import React from "react"
+import * as request from "request"
+import styled, { StyledFunction } from "styled-components"
+import Colors from "../../assets/colors"
+import InvertedButton from "../buttons/inverted"
+import { borderedInput } from "../mixins"
+import Fonts from "./fonts"
+
+interface EmailSignupProps {
+  signupUrl: string
+}
+
+interface EmailSignupState {
+  value: string
+  error: boolean
+  submitted: boolean
+  disabled: boolean
+  message: string
+}
+
+interface InputProps {
+  isError: boolean
+  isReadOnly: boolean
+}
+
+class EmailSignup extends React.Component<EmailSignupProps, EmailSignupState> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: "",
+      error: false,
+      submitted: false,
+      disabled: false,
+      message: "",
+    }
+  }
+
+  onClick = () => {
+    this.setState({ disabled: true })
+    request.post(
+      {
+        uri: this.props.signupUrl,
+        email: this.state.value,
+      },
+      err => {
+        if (err) {
+          this.setState({ message: "Invalid Email... Please try again", error: true })
+          setTimeout(() => {
+            this.setState({ message: "", disabled: false, error: false })
+          }, 2000)
+        } else {
+          this.setState({ message: "Thank you!" })
+          setTimeout(() => {
+            this.setState({ message: "", disabled: false, submitted: true })
+          }, 2000)
+        }
+      }
+    )
+  }
+
+  onInputChange = e => {
+    this.setState({ value: e.target.value, error: null })
+  }
+
+  render() {
+    if (this.state.submitted) {
+      return <div />
+    } else {
+      return (
+        <EmailSignupContainer>
+          <Title>Stay up to date with Artsy Editorial</Title>
+          <Form>
+            <Input
+              type="email"
+              placeholder="Enter Your Email..."
+              onChange={this.onInputChange}
+              value={this.state.message || this.state.value}
+              readOnly={this.state.message.length > 0}
+              isError={this.state.error}
+              isReadOnly={this.state.message.length > 0}
```

Comment:
Seems a little confusing here, but we're making use of `value` and `readOnly` to display a message, whether it be a success or error.